src/main/java/EquationServlet.java

```diff
@@ -0,0 +1,118 @@
+        Map<String, String> StringValue = new HashMap<>();
+        for (Map.Entry<String, String[]> par : req.getParameterMap().entrySet()) {
+            if (par.getKey().equals("equation")) continue;
+            try {
+                args.put(par.getKey(), Integer.parseInt(par.getValue()[0]));
+            } catch (NumberFormatException e) {
+                StringValue.put(par.getKey(), par.getValue()[0]);
+            }
+        }
+        for (String key : StringValue.keySet()) {
+            Integer value = args.get(StringValue.get(key));
+            args.put(key, value);
+        }
+        List<String> polishTokens = parseIntoPolishNotation(equation);
+        int result = calculate(polishTokens, args);
+        resp.getOutputStream().print(result);
+        resp.getOutputStream().flush();
+        resp.getOutputStream().close();
+    }
+
+    private List<String> parseIntoPolishNotation(String equation) {
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation.replaceAll(" ", ""), "()+-/*", true);
+        while (tokenizer.hasMoreElements()) {
+            tokens.add(tokenizer.nextToken());
+        }
+        Deque<String> stack = new ArrayDeque<>();
+        List<String> polishNotation = new ArrayList<>();
+        for (String token : tokens) {
+            try {
+                switch (token) {
+                    case "+":
+                    case "-":
+                    case ")":
+                        while ("+-/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        if (token.equals(")")) {
+                            stack.removeFirst();
+                        } else {
+                            stack.addFirst(token);
+                        }
+                        break;
+                    case "*":
+                    case "/":
+                        while ("/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        stack.addFirst(token);
+                        break;
+                    case "(":
+                        stack.addFirst(token);
+                        break;
+                    default:
+                        polishNotation.add(token);
+                }
+            } catch (NoSuchElementException e) {
+                if (!token.equals(")"))
+                    stack.addFirst(token);
+            }
+        }
+        while (!stack.isEmpty()) {
+            polishNotation.add(stack.removeFirst());
+        }
+        return polishNotation;
+    }
+
+    private static int calculate(List<String> polishNotation, Map<String, Integer> args) {
+        Deque<Integer> stack = new ArrayDeque<>();
+        for (String token : polishNotation) {
+            int first;
+            int second;
+            switch (token) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4302155)